File: 2020-10-30-client-libraries/php/idempotency/create_customer.php
<?php

// For more information about this demo, please see:
//   https://www.youtube.com/watch?v=xmfgetuwf-0

require_once('./vendor/autoload.php');

$stripe = new \Stripe\StripeClient(
  'sk_test_xxx'
);

$stripe->customers->create([
  'email' => 'foo@bar.com',
], ['idempotency_key' => 'a-long-random-string-20210115164426']);

File: 2020-10-30-client-libraries/go/idempotency/create_customer.go
package main

import (
  "github.com/stripe/stripe-go/v72"
  "github.com/stripe/stripe-go/v72/customer"
)

func main() {
  stripe.Key = "sk_test_xxx"

  params := &stripe.CustomerParams{
    Email: stripe.String("foo@bar.com"),
  }
  params.SetIdempotencyKey("a-long-random-string-20201014155429")

  customer.New(params)
}
